gqlalchemy/utilities.py

```diff
@@ -106,7 +109,20 @@ def to_cypher_value(value: Any, config: NetworkXCypherConfig = None) -> str:
         return str(value)
 
     if isinstance(value, (timedelta, time, datetime, date)):
-        return f"{datetimeKwMapping[value_type]}('{_format_timedelta(value) if isinstance(value, timedelta) else value.isoformat()}')"
+        if isinstance(value, datetime):
+            if value.tzinfo == pytz.UTC:
+                formatted_date = value.strftime(f"%Y-%m-%dT%H:%M:%SZ")
+                return f"{DatetimeKeywords.ZONEDDATETIME.value}('{formatted_date}')"
+            elif value.tzinfo is not None:
+                tz = value.strftime('%z')
+                tz = f"{tz[:3]}:{tz[3:]}" 
+                tz_name = value.tzinfo.zone
+                formatted_date = value.strftime(f"%Y-%m-%dT%H:%M:%S{tz}")
+                return f"{DatetimeKeywords.ZONEDDATETIME.value}('{formatted_date}[{tz_name}]')"
+            else:
+                return f"{DatetimeKeywords.LOCALDATETIME.value}('{value.isoformat()}')"
+        else:
+            return f"{datetimeKwMapping[value_type]}('{_format_timedelta(value) if isinstance(value, timedelta) else value.isoformat()}')"
```

Comment:
Avoid using nested isinstance, logically not the best.

Do `isinstance` checks on the same if-else level

```
if is datetime:
....
elif is timedelta;
elif is ...:
else

instead of:
if isinstance ...:
  if isinstance ...

```